t/plugin/clickhouse-logger.t

```diff
@@ -0,0 +1,227 @@
+                                                 logtable = "t",
+                                                 endpoint_addr = "http://127.0.0.1:8123",
+                                                 max_retry_count = 1,
+                                                 name = "clickhouse logger",
+                                                 ssl_verify = false
+                                                 })
+
+            if not ok then
+                ngx.say(err)
+            else
+                ngx.say("passed")
+            end
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 2: Basic configuration verification
+--- yaml_config
+plugins:
+  - clickhouse-logger
+--- config
+    location /t {
+        content_by_lua_block {
+            local plugin = require("apisix.plugins.clickhouse-logger")
+            local ok, err = plugin.check_schema({user = "default",
+                                                 password = "a",
+                                                 database = "default",
+                                                 logtable = "t",
+                                                 endpoint_addr = "http://127.0.0.1:8123"
+                                                 })
+
+            if not ok then
+                ngx.say(err)
+            else
+                ngx.say("passed")
+            end
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 3: auth configure undefined
+--- yaml_config
+plugins:
+  - clickhouse-logger
+--- config
+    location /t {
+        content_by_lua_block {
+            local plugin = require("apisix.plugins.clickhouse-logger")
+            local ok, err = plugin.check_schema({user = "default",
+                                                 password = "a",
+                                                 database = "default",
+                                                 logtable = "t"
+                                                 })
+
+            if not ok then
+                ngx.say(err)
+            else
+                ngx.say("passed")
+            end
+        }
+    }
+--- response_body
+property "endpoint_addr" is required
+
+
+
+=== TEST 4: add plugin on routes
+--- yaml_config
+apisix:
+    node_listen: 1984
+    admin_key: null
+plugins:
+  - clickhouse-logger
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                        "plugins": {
+                            "clickhouse-logger": {
+                                "user": "default",
+                                "password": "a",
+                                "database": "default",
+                                "logtable": "t",
+                                "endpoint_addr": "http://127.0.0.1:8123"
```

Comment:
Let's configure to use the fake upstream and check the upload log data via grep_error_log_out

Reply:
ok

Reply:
wonderful, it works.